src/process/ProcessManagerImpl.cpp

```diff
@@ -726,6 +761,18 @@ ProcessManagerImpl::maybeRunPendingProcesses()
     {
         auto i = mPending.front();
         mPending.pop_front();
+
+        if (fs::exists(i->mImpl->mOutFile))
+        {
+            i->mImpl->cancel(asio::error_code(asio::error::invalid_argument,
```

Comment:
the better way (avoids duplicating code and logs properly) is to just throw a `runtime_error` with that error message